web/template_manager.go

```diff
@@ -0,0 +1,71 @@
+package web
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"io/fs"
+	"path/filepath"
+)
+
+const (
+	base       = "templates/base.html.tmpl"
+	blocksHtml = "templates/blocks/*.html.tmpl"
+)
+
+// the main motivation of this code is to parse all templates contained in templates directory,
+// respecting the "base" template which is the basis, where other custom templates will inherit and override content
+// we also extend the template and build the base template with blocks templated (contained in /templates/blocks directory)
+
+// AddAllTemplateFromFs return a map of parsed templated from filesystem
+// the map of parsed html templates  is consumed by http handler, accessed via key
+func AddAllTemplatesFromFS(templatesFS fs.FS, templates ...string) map[string]template.Template {
+	templs := make(map[string]template.Template)
+
+	for _, pattern := range templates {
+		files, err := fs.Glob(templatesFS, pattern)
+		if err != nil {
+			// we exceptionally hard panic in case of glob errors, these should never happen.
+			panic(err)
+		}
+		for _, file := range files {
+			if file == base {
+				continue
+			}
+			addFileFromFS(templatesFS, file, templs)
+		}
+	}
+	return templs
+}
+
+// addFileFromFS parses the base template with the user
+func addFileFromFS(templatesFS fs.FS, file string, templs map[string]template.Template) {
```

Comment:
A possible dumb question. Is there any reason why this `addFileFromFS` starts with lower case and `AddAllTemplatesFromFS` with upper case?
Does this mean that the 1st is private and the 2nd public?

Reply:
yes this is exported in golang

Reply:
@arbulu89 precisely. As @MalloZup mentioned, in golang everything is either "exporter" or not, from a package. You can use non-exported symbols within the same package, but they must be exported to be used in other packages; bear in mind that there is no package hierarchy, i.e. even if you can nest packages with directories the package hierarchy gets flattened, so to use a symbol in a pseudo-subpackage, you need to export it.